addchain/genesis.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+func createExtraGenesisData(superchainRepoPath, superchainTarget, monorepoDir, genesisConfig, chainName string) error {
+	// Create directory for genesis data
+	genesisDir := filepath.Join(superchainRepoPath, "superchain", "extra", "genesis", superchainTarget)
+	if err := os.MkdirAll(genesisDir, 0755); err != nil {
+		return fmt.Errorf("failed to create genesis directory: %w", err)
+	}
+
+	// Set current working directory to the monorepo directory
+	if err := os.Chdir(monorepoDir); err != nil {
+		return fmt.Errorf("failed to change directory to monorepo: %w", err)
+	}
+
+	cmd := exec.Command("go", "run", "./op-chain-ops/cmd/registry-data",
+		"--l2-genesis", genesisConfig,
+		"--bytecodes-dir", filepath.Join(superchainRepoPath, "superchain", "extra", "bytecodes"),
+		"--output", filepath.Join(genesisDir, chainName+".json.gz"))
+
```

Comment:
Should we consider putting this new package into the `registry-data` command? If we did so, we might be freer to use libraries such as go-ethereum in order to get the job done. 

We want to avoid pulling such dependencies into the `superchain` module, since that is upstream of `op-geth`.